frontend/lib/identification/verification_workflow/dialogs/remove_card_confirmation_dialog.dart

```diff
@@ -0,0 +1,98 @@
+          {required BuildContext context,
+          required DynamicUserCode userCode,
+          required CarouselController carouselController}) =>
+      showDialog(
+        context: context,
+        builder: (_) => RemoveCardConfirmationDialog(userCode: userCode, carouselController: carouselController),
+      );
+
+  @override
+  RemoveCardConfirmationDialogState createState() => RemoveCardConfirmationDialogState();
+}
+
+class RemoveCardConfirmationDialogState extends State<RemoveCardConfirmationDialog> {
+  @override
+  Widget build(BuildContext context) {
+    final localization = buildConfig.localization.identification.removeCardDialog;
+    final projectId = Configuration.of(context).projectId;
+    final regionsQuery = GetRegionsByIdQuery(
+      variables: GetRegionsByIdArguments(
+        project: projectId,
+        ids: [widget.userCode.info.extensions.extensionRegion.regionId],
+      ),
+    );
+
+    return Query(
+      options: QueryOptions(document: regionsQuery.document, variables: regionsQuery.getVariablesMap()),
+      builder: (result, {refetch, fetchMore}) {
+        final data = result.data;
+        final theme = Theme.of(context);
+        final region = result.isConcrete && data != null ? regionsQuery.parse(data).regionsByIdInProject[0] : null;
+        return AlertDialog(
+          titlePadding: EdgeInsets.all(4),
+          contentPadding: EdgeInsets.only(left: 20, right: 20),
+          actionsPadding: EdgeInsets.only(left: 20, right: 20, top: 10, bottom: 10),
+          title: ListTile(
+            leading: Icon(Icons.warning, color: theme.colorScheme.primaryContainer, size: 30),
+            title: Text(localization.title, style: TextStyle(fontSize: 18)),
+          ),
+          content: SingleChildScrollView(
+            child: ListBody(
+              children: <Widget>[
+                Padding(
+                    padding: EdgeInsets.only(bottom: 20),
+                    child: Text(localization.description, style: TextStyle(fontSize: 14))),
+                IdCard(
+                  cardInfo: widget.userCode.info,
+                  region: region != null ? Region(region.prefix, region.name) : null,
+                  // We trust the backend to have checked for expiration.
+                  isExpired: false,
+                  isNotYetValid: false,
+                ),
+              ],
+            ),
+          ),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Abbrechen'),
+              onPressed: () {
+                Navigator.of(context).pop(false);
+              },
+            ),
+            TextButton(
+              child: const Text('Löschen'),
+              onPressed: () {
+                final provider = Provider.of<UserCodeModel>(context, listen: false);
+                if (provider.userCodes.length == 1) {
+                  provider.removeCodes();
+                } else {
+                  provider.removeCode(widget.userCode);
```

Comment:
```suggestion
                provider.removeCode(widget.userCode);
                if (provider.userCodes.length != 1) {
 ```

Reply:
this is not the same. Now you would remove all codes if the length is not 1.
I just ensure for the last code removed that its reseted and not including any chars like separators or thx